bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/internal/LinkTapHandler.java

```diff
@@ -0,0 +1,502 @@
+
+    private static final double GW_LITRES_M3_CONVERSION_RATIO = 3.785;
+
+    private final Storage<String> strStore;
+
+    private static final List<String> READBACK_DISABLED_CHANNELS = List.of(DEVICE_CHANNEL_OH_VOLUME_LIMIT,
+            DEVICE_CHANNEL_OH_DURATION_LIMIT);
+
+    private java.util.concurrent.@Nullable ScheduledFuture<?> pausePlanFuture = null;
+    private final Object pausePlanLock = new Object();
+    private volatile boolean pausePlanActive = false;
+
+    public LinkTapHandler(Thing thing, Storage<String> strStore) {
+        super(thing);
+        this.strStore = strStore;
+    }
+
+    /**
+     * Abstract method implementations from PollingDeviceHandler
+     * required for the lifecycle of LinkTap devices.
+     */
+
+    @Override
+    protected void runStartInit() {
+        try {
+            if (config.enableAlerts) {
+                sendRequest(new AlertStateReq(0, true));
+            }
+
+            final String[] chansToRefresh = new String[] { DEVICE_CHANNEL_PAUSE_PLAN_OVERRIDE,
+                    DEVICE_CHANNEL_PAUSE_PLAN_EXPIRES, DEVICE_CHANNEL_OH_DURATION_LIMIT,
+                    DEVICE_CHANNEL_OH_VOLUME_LIMIT };
+            for (String chanId : chansToRefresh) {
+                final Channel pausePlanChan = getThing().getChannel(chanId);
+                if (pausePlanChan != null) {
+                    handleCommand(pausePlanChan.getUID(), RefreshType.REFRESH);
+                }
+            }
+
+            final String pausePlanState = strStore.get(DEVICE_CHANNEL_PAUSE_PLAN_OVERRIDE);
+            if (OnOffType.ON.toString().equals(pausePlanState)) {
+                scheduleRenewPlanPause();
+            }
+
+        } catch (final InvalidParameterException ipe) {
+            logger.warn("Raise Bug Report: {} - Failed to enable all alerts - invalid parameter exception",
+                    getThing().getLabel());
+        }
+    }
+
+    @Override
+    protected void registerDevice() {
+        LinkTapBridgeHandler.DEV_ID_LOOKUP.registerItem(registeredDeviceId, this, () -> {
+        });
+    }
+
+    @Override
+    protected void deregisterDevice() {
+        LinkTapBridgeHandler.DEV_ID_LOOKUP.deregisterItem(registeredDeviceId, this, () -> {
+        });
+    }
+
+    @Override
+    protected String getPollResponseData() {
+        try {
+            return sendRequest(new DeviceCmdReq(CMD_UPDATE_WATER_TIMER_STATUS));
+        } catch (final InvalidParameterException ipe) {
+            logger.warn("Raise Bug Report: {} - Poll failure - invalid parameter exception", getThing().getLabel());
+            return "";
+        }
+    }
+
+    @Override
+    protected void processPollResponseData(String data) {
+        processCommand3(data);
+    }
+
+    /**
+     * OpenHab handlers
+     */
+
+    @Override
+    public void dispose() {
+        cancelPlanPauseRenew();
+        super.dispose();
+    }
+
+    private void scheduleRenewPlanPause() {
+        synchronized (pausePlanLock) {
```

Comment:
See other comments about locking, what case do you see that is fixed by this lock?

Reply:
Replied to other one :)